### ISSUE TITLE
Report malformed custom policies when external checks is enabled

### ISSUE THREAD

**Issue by SaltwaterC** [bridgecrewio/checkov#4288]
This is something that caught me as a Checkov newb. Basically, it boils down to an editor auto-indenting failure. I know, I'm holding it wrong.

What I tried to do:

```yaml
metadata:
  [...]
definition:
  [...]
```

What my editor has done by trying to be overly helpful when pasting a new definition despite removing all space from the line following the metadata structure:

```yaml
metadata:
  [...]
  definition:
    [...]
```

This went unnoticed. Checkov essentially ignores a custom policy with missing definition or when the definition is attached to the metadata object. This has lead to a lot of head scratching until visually identifying the failure. The fact that `-f` and Terraform plan files had problems in the past regarding custom checks, the question of a regression definitely popped up. Turns out, that was a red herring.

It would be extremely helpful if custom policies with missing definitions are reported as such rather than ignored. Thanks!

**Comment by gruebel** [comment]
hey @SaltwaterC thanks for reaching out.
I actually started to work on a better error reporting of misconfigured custom checks. What would you except to happen? Is a log message with log level warning enough or should be error level? Or do you have something completely different in mind?

**Comment by SaltwaterC** [comment]
Hey. Thanks for getting back to me. I think anything would work. I believe this should be an error, but I can live with a warning. At least that would give some indication that something isn't right.
